src/rpc/blockchain.cpp

```diff
@@ -1924,6 +1924,28 @@ static UniValue savemempool(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static UniValue clearmempool(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 0) {
+        throw std::runtime_error(
+            "clearmempool\n"
+            "\nRemoves all transactions from the mempool.\n"
+            "It will fail if a previous dump is still being loaded.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("clearmempool", "")
+            + HelpExampleRpc("clearmempool", "")
+        );
+    }
+
+    if (!g_is_mempool_loaded) {
+        throw JSONRPCError(RPC_MISC_ERROR, "The mempool was not loaded yet");
+    }
+
+    mempool.clear();
```

Comment:
Are we missing some lock acquisitions here (e.g. `mempool.cs`)?

Reply:
I thought so first myself, but `CTxMempool::clear` explicitly locks for you.